Reflection/CustomConfigurationDemo/Providers/ConfigurationManagerConfigurationProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Serializer = Newtonsoft.Json.JsonSerializer;
+
+namespace CustomConfigurationDemo.Providers;
+
+public class ConfigurationManagerConfigurationProvider : IConfigurationProvider
+{
+    private readonly string _appSettingsPath = Path.Combine(Environment.CurrentDirectory, "appsettings.json");
```

Comment:
Well, ideally, all paths to the files should be passed via constructor since it can be dynamically changed, you can use fallback with this expression if it was not passed.
Let's try to use 2 constructors overloading: default one and overloading with `appSettingsPath` in constructor. It will give you much more flexibility